SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityJobPoller.java

```diff
@@ -69,6 +72,16 @@ public void runActionOnPoll() {
         continue;
       }
 
+      if (start - taskId.getStartedAt() >= configuration.getTaskExecutionTimeout()) {
+        taskManager.createTaskCleanup(new SingularityTaskCleanup(
+            Optional.<String>absent(),
+            TaskCleanupType.DEPLOY_TIMED_OUT,
+            start,
+            taskId,
+            Optional.of("Deploy has reached/exceeded the set timeout"),
```

Comment:
Again, the "deploy" wording isn't quite right here. Also, consider updating the message to include the actual timeout value (i.e. something like `Task has exceeded maximum execution time of XX:XX`)